### PR TITLE
Fixes #965 : Corrected the adapter complying with backend api

### LINKED ISSUE #965: Correct the adapter complying with backend api
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behaviour:**
The adapter is not correctly set up complying with the backend api

**Expected behaviour:**
The adapter should be set according to the backend api.
**Steps to reproduce:**

**yes** I am working on it

### PR DESCRIPTION
**Pull request by abhi40308**
Fixes #965 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-966-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:
- Corrected adapter according to the backend api
- Patch for the generate badges endpoints
-

### CODE REVIEW

**Comment by abhi40308** [comment]
@gabru-md @ParthS007 @yashLadha  Please review

File: frontend/app/adapters/def-image-upload.js [hunk 1]
```diff
@@ -2,9 +2,9 @@ import DS from 'ember-data';
 import ENV from '../config/environment';
 
 const { APP } = ENV;
-const { RESTAdapter } = DS;
+const { JSONAPIAdapter } = DS;
```
Comment (duttaditya18) [inline frontend/app/adapters/def-image-upload.js]: Why this change?

Comment (yashLadha) [reply frontend/app/adapters/def-image-upload.js]: This is because the backend api needs the data to be complying with JSON spec an not REST spec, as ember data treats them both differently.

Comment (duttaditya18) [reply frontend/app/adapters/def-image-upload.js]: Oh! All right! 👍
